fem/pgridfunc.cpp

```diff
@@ -481,6 +481,27 @@ void ParGridFunction::GetVectorValue(ElementTransformation &T,
    }
 }
 
+void ParGridFunction::GetDerivative(int comp, int der_comp,
+                                    ParGridFunction &der)
+{
+   Array<int> overlap;
+   AccumulateAndCountDerivativeValues(comp, der_comp, der, overlap);
+
+   // Count the zones globally.
+   GroupCommunicator &gcomm = der.ParFESpace()->GroupComm();
+   gcomm.Reduce<int>(overlap, GroupCommunicator::Sum);
+   gcomm.Bcast(overlap);
+
+   // Accumulate for all dofs.
+   gcomm.Reduce<double>(der.GetData(), GroupCommunicator::Sum);
```

Comment:
Can you update these GetData calls to being `HostReadWrite()` rather than `GetData()`? Alternatively, could you add a `HostReadWrite()` call to the `AccumulateAndCountDerivativeValues` call for `GridFunction`. I just want to make sure the data is actually on the host before we make these calls and the user doesn't end up having code failing in odd places once they start making use of the GPU.

Reply:
I just say this because I've run into this issue before with the `(Par)GridFunction::ProjectDiscCoefficient` functions and that was not the easiest issue to track down.

Reply:
Done. See also #2581.